### DISCUSSION TITLE
How to remove table name prefix from column names?

### DISCUSSION THREAD

**Discussion by juancarloslm** [sqlalchemy/sqlalchemy#7252] [Usage Questions] [answered]
`Python version: 3.7.4`
`sqlalchemy version: 1.3.18`

I completed the migration of an API from Flask to fastapi; but, for some reason the keys of the json response in fastapi are prefixed with the table name. 

This is the function to get the results from the database:
```python
async def get_one_by_id(db_session, id: int):
        query = Query(Model).filter_by(id=id)
        results = db_session.execute(query)
        return results.fetchone()
```

When I call the method `.keys()` on the results, I get this:
`['model_id', 'model_external_key', 'model_name', 'model_description', 'model_tags', 'model_version', 'model_created_by', 'model_created_at', 'model_updated_at']`

but I expect to get this:
`['id', 'external_key', 'name', 'description', 'tags', 'version', 'created_by', 'created_at', 'updated_at']`

The column names in the database doesn't have any prefix, that's why I'm wondering if there is a configuration that I'm missing or something else.
I already spent all day reading the documentation and searching for a solution, but I didn't find anything, maybe I'm searching wrong.
Also, I already tried using the `__mapper_args__ = {"column_prefix":""}` in the Model class.

This is how the database session is declared:
```python
from typing import Generator

from sqlalchemy import create_engine, MetaData
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import sessionmaker

from config import Settings

settings = Settings.get_settings()

database_url = f"postgresql://{settings.db_user}:{settings.db_password}@{settings.db_host}:{settings.db_port}/{settings.db_name}"
engine = create_engine(database_url)
SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)

def get_db() -> Generator:
    try:
        db = SessionLocal()
        yield db
    finally:
        db.close()

Base = declarative_base()
```
This is how the model is declared:
```python
import sqlalchemy as sa

from app.endpoints._public.v1.db import Base, Presets

class Model(Base):
    __tablename__ = "model"

    id = Presets.primary_key()
    external_key = Presets.string_token()
    name = sa.Column(sa.String(100), nullable=False)
    description = sa.Column(sa.Text, nullable=True)
    tags = sa.Column(sa.String(150), nullable=True)
    version = sa.Column(sa.Float, nullable=False)
    run_id = sa.Column(sa.String(150), sa.ForeignKey("run.id"), nullable=True)
    created_by = sa.Column(sa.String(150), nullable=False)
    created_at = Presets.timestamptz(nullable=False)
    updated_at = Presets.timestamptz()
```

Thank you very much in advance!

**Comment by zzzeek** [comment, marked as answer]
if you run your program with `python -Werror` you will get this error:

```
sqlalchemy.exc.SADeprecationWarning: Object <sqlalchemy.orm.query.Query object at 0x7fdb9f5b2ca0> should not be used directly in a SQL statement context, such as passing to methods such as session.execute().  This usage will be disallowed in a future release.  Please use Core select() / update() / delete() etc. with Session.execute() and other statement execution methods.

```

use the select() construct for composing ORM queries to be passed to session.execute().   session.query() is fully legacy and has no support within the async API.

**Reply by CaselIT** [reply]
It uses version 1.3.18, so that's not supported.

But you should use `db_session.query(Model).filter_by(id=id).first()`. If id is the primary key you can also just use `db_session.query(Model).get(id)`.

As mike mentioned, version 1.3 has no support for async, so each time you call the db you are blocking the loop (and the application)

**Comment by juancarloslm** [comment]
Thanks zzzeek and CaselIT! Both methods work as expected. (Yes, for some reason I have no problem using `select` )
Then, the problem is how the query is built or executed.

**Reply by CaselIT** [reply]
also consider updating to 1.4 and using an async driver so that you can properly use asyncio without blocking for the queries
